pkgs/tools/admin/trivy/default.nix

```
@@ -29,9 +32,22 @@ buildGoModule rec {
     "-X=github.com/aquasecurity/trivy/pkg/version.ver=v${version}"
   ];
 
+  nativeBuildInputs = [ installShellFiles ];
+
   # Tests require network access
   doCheck = false;
 
+  postInstall =
+    let
+      trivy = if stdenvNoCC.buildPlatform.canExecute stdenvNoCC.hostPlatform then "\"$out\"" else buildPackages.trivy;
+    in
+    ''
+      installShellCompletion --cmd ${pname} \
+        --bash <(${trivy}/bin/${pname} completion bash) \
+        --fish <(${trivy}/bin/${pname} completion fish) \
+        --zsh <(${trivy}/bin/${pname} completion zsh)
```

```suggestion
      installShellCompletion --cmd trivy \
        --bash <(${trivy}/bin/trivy completion bash) \
        --fish <(${trivy}/bin/trivy completion fish) \
        --zsh <(${trivy}/bin/trivy completion zsh)
```
the assumption that the binary name must match the package name is incorrect